resharper/resharper-unity/src/Yaml/Psi/Modules/UnityExternalFilesModuleProcessor.cs

```diff
@@ -83,11 +90,8 @@ public void OnUnityProjectAdded(Lifetime projectLifetime, IProject project)
             // Note that this means we process meta and asset files for class library projects, which likely won't have
             // asset files. We can't use IsUnityGeneratedProject because any project based in the Packages folder or
             // using 'file:' won't be processed.
-            if (!myAssetSerializationMode.IsForceText || !myYamlSupport.IsParsingEnabled.Value ||
-                !project.IsUnityProject())
-            {
+            if (!project.IsUnityProject())
```

Comment:
We shouldn't walk the file system if asset serialisation mode is not set to "ForceText"

Reply:
This allows us to handle change of serialisation mode in Unity without solution reopenning.

Why do you think, that adding psiSourceFiles can affect performance?